src/lib/journald.ts

```diff
@@ -0,0 +1,138 @@
+import { ChildProcess, spawn } from 'child_process';
+import { EventEmitter } from 'events';
+import { isArray } from 'lodash';
+import StrictEventEmitter from 'strict-event-emitter-types';
+
+import log from './supervisor-console';
+
+const CHROOT_COMMAND = 'chroot';
+const CHROOT_DIRECTORY = '/mnt/root';
+const JOURNALD_COMMAND = 'journalctl';
+const JOURNALD_FORMAT = 'json';
+const JOURNALD_FORMAT_FLAG = '-o';
+const JOURNALD_FOLLOW_FLAG = '-f';
+const JOURNALD_SERVICE_NAME_FLAG = '-u';
+const JOURNALD_COUNT_FLAG = '-n';
+
+export interface Events {
+	logLine: {
+		message: string;
+		priority: number;
+		name: string;
+		timestamp: Date;
+
+		// We also forward all other fields which we get
+		[key: string]: unknown;
+	};
+	stderrLine: string;
+	disconnect: { code: number; signal: string };
+	error: Error;
+}
+
+export type JournaldEventEmitter = StrictEventEmitter<EventEmitter, Events>;
+
+export class Journald extends (EventEmitter as new () => JournaldEventEmitter) {
```

Comment:
Should not it also be "Closable", so that when HTTP connection is closed, we can close the pipe on our side?